sharktank/sharktank/ops/attention_impls.py

```diff
@@ -7,25 +7,88 @@
 """Implementations for op variants that are fully quantized.
 """
 
-from types import NoneType
 import math
 import torch
+import warnings
+from types import NoneType
 
-from sharktank import kernels
+from sharktank import kernels, ops
 from sharktank.types import (
     AnyTensor,
     PlanarQuantizedTensor,
 )
-from sharktank import kernels
 
 from sharktank.types.layouts import TensorScaledLayout
 
-from sharktank.utils import debugging
-
-from sharktank.types.tensors import ReplicatedTensor, unbox_tensor
+from sharktank.types.tensors import unbox_tensor
 from .signatures import (
     scaled_dot_product_attention,
 )
+from ._registry import AnyOfType
+
+# These two versions should be preserved in this order
+@scaled_dot_product_attention.override(
+    AnyTensor,
+    AnyTensor,
+    AnyTensor,
+    None,
```

Comment:
Why is the mask forced to always be `None`? There is logic afterwards that deal with both cases.